sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3419,6 +3419,39 @@ class Dataset[T] private[sql](
     } (Encoders.STRING)
   }
 
+  /**
+   * Returns the content of the Dataset as a Dataset of JSON strings.
+   * @since 3.0.1
+   */
+  def toJSON(options: Map[String, String]): Dataset[String] = {
```

Comment:
The built-in `to_json` is not enough for your usecase? cc: @HyukjinKwon

Reply:
I guess so, you'd have to do something like this `df.select(to_json(struct("*"), Map("ignoreNullFields" -> "false")).as("value"))` to produce the same result as `df.toJSON(options)`

Reply:
Yea, I think so.

Reply:
I don't feel strongly here. I would rather don't extend `toJSON` anymore given that we added `to_json`, `from_json`, etc. to cover such general cases. I am okay if any committer supports it.

Reply:
Thanks for the check, @HyukjinKwon. I'm neutral on this, too.